### PR TITLE
Fixes vatbeast slapping. 

### PR DESCRIPTION
**Pull request by Krysonism**
## About The Pull Request

This fixes a bug that caused the vat beast slap ability to runtime and fail.

This was caused by the owner variable on the proc holder never being set.

closes:https://github.com/tgstation/tgstation/issues/56753

## Why It's Good For The Game

Vatbeasts can finally slap again after 10 months, this is probably the biggest reason to make the mob, so it is essential that it works properly.

bild

## Changelog
:cl:
fix: Vatbeasts can use their powerful tentacle slap ability again.
/:cl:

### CODE REVIEW

File: code/modules/spells/spell.dm [hunk 1]
```diff
@@ -16,9 +16,9 @@
 	var/base_action = /datum/action/spell_action
 	var/mob/living/owner
 
-/obj/effect/proc_holder/Initialize(mob/living/owner)
+/obj/effect/proc_holder/Initialize(mapload, mob/living/new_owner)
 	. = ..()
-	src.owner = owner
+	owner = new_owner
```
Comment (Watermelon914) [inline code/modules/spells/spell.dm:21]: wouldn't this be better as a weakref? It leaves a reference to a mob without checking for when it is deleted. At the very least, you might want to register a signal on when the owner gets qdeleted so you can remove them as a reference.

Comment (Krysonism) [reply code/modules/spells/spell.dm:21]: :pensive:

File: code/modules/spells/spell.dm [hunk 2]
```diff
@@ -14,11 +14,11 @@
...
 	var/action_background_icon_state = "bg_spell"
 	var/base_action = /datum/action/spell_action
-	var/mob/living/owner
+	var/datum/weakref/owner
 
-/obj/effect/proc_holder/Initialize(mob/living/owner)
+/obj/effect/proc_holder/Initialize(mapload, mob/living/new_owner)
 	. = ..()
-	src.owner = owner
+	owner = WEAKREF(new_owner)
```
Comment (coiax) [inline code/modules/spells/spell.dm:21]: So nothing else uses this proc_holder `owner` var?

Comment (Krysonism) [reply code/modules/spells/spell.dm:21]: I am pretty sure it is just the vat beast. 

Floyd added it to be able to relay abilities to the rider.

**Pull request merged**